install_sct

```diff
@@ -218,74 +234,77 @@ if [[ -n \"\$BASH_VERSION\" ]]; then
-    if [[ -a $profiles ]]; then
-      if ! grep -E "(\.|source) .*bashrc" $profiles >/dev/null 2>&1; then
-        echo "$sourceblock" >>$profiles
+  bidon=""
+  for profile in ~/.bash_profile ~/.bash_login ~/.profile; do
+    if [[ -a "$profile" ]]; then
+      if ! grep -E "(\.|source) .*bashrc" "$profile" >/dev/null 2>&1; then
+        echo "$sourceblock" >>"$profile"
       fi
-      bidon=0
+      bidon="done"
       break
     fi
   done
 
-  if [[ -z $bidon ]]; then
+  if [[ -z "$bidon" ]]; then
     echo "$sourceblock" >>~/.bash_profile
   fi
 }
 
 # Installation text to insert in shell config file
 function edit_shellrc() {
   # Write text common to all shells
-  echo
-  echo "" >>$RC_FILE_PATH
-  echo "# SPINALCORDTOOLBOX (installed on $(date +%Y-%m-%d\ %H:%M:%S))" >>$RC_FILE_PATH
-  echo $DISPLAY_UPDATE_PATH >>$RC_FILE_PATH
-  # Switch between shell
-  if [[ $THE_RC == "bash" ]]; then
-    echo "export SCT_DIR=$SCT_DIR" >>$RC_FILE_PATH
-    echo "export MPLBACKEND=Agg" >>$RC_FILE_PATH
-  elif [[ $THE_RC == "csh" ]]; then
-    echo "setenv SCT_DIR $SCT_DIR" >>$RC_FILE_PATH
-    echo "setenv MPLBACKEND Agg" >>$RC_FILE_PATH
-  fi
-  # add line
-  echo "" >>$RC_FILE_PATH
+  (
+    echo
+    echo ""
+    echo "# SPINALCORDTOOLBOX (installed on $(date +%Y-%m-%d\ %H:%M:%S))"
+    echo "$DISPLAY_UPDATE_PATH"
+    # Switch between shell
+    if [[ "$THE_RC" == "bash" ]]; then
+      echo "export SCT_DIR=$SCT_DIR"
+      echo "export MPLBACKEND=Agg"
+    elif [[ "$THE_RC" == "csh" ]]; then
+      echo "setenv SCT_DIR $SCT_DIR"
+      echo "setenv MPLBACKEND Agg"
+    fi
+    # add line
+    echo ""
+  ) >> "$RC_FILE_PATH"
 }
 
 # Download from URL using curl/wget
 function download() {
-  # Use curl or wget to download goodies
-  e_status=0
-  # Try with wget
-  if [[ $(which wget) ]]; then
-    cmd="wget -O $1 $2"
-    print code "$cmd"
-    $cmd
-    e_status=$?
-    echo exit status is $e_status
-  fi
-  # Try with curl
-  if [[ $(which curl) && ! -e $1 ]]; then
-    cmd="curl -o $1 -L $2"
-    print code "$cmd"
-    $cmd
-    e_status=$?
-    echo exit status is $e_status
-  fi
-  # check success
-  if [[ $e_status -ne 0 || ! -e $1 ]]; then
-    die "The download of $2 failed\n
-Please check your internet connection before relaunching the installer\n"
-  fi
+  ((command -v wget >/dev/null) && run wget -O "$1" "$2" ) ||
+  ((command -v curl >/dev/null) && run curl -o "$1" -L "$2" ) ||
+  die "The download of $2 failed
+Please check that you have wget or curl installed, and
+your internet connection before relaunching the installer"
 }
 
 # Usage of this script
 function usage() {
-  echo -e "\nUsage: $0 [-d] [-b] [-v]" 1>&2
-  echo -e "\nOPTION"
-  echo -e "\t-d \v Prevent the (re)-installation of the \"data/\" directory "
-  echo -e "\n\t-b \v Prevent the (re)-installation of the SCT binaries files "
-  echo -e "\n\t-v \v Full verbose"
+  # extract the usage block from our own header
+  awk '
```

Comment:
This is really nifty. Slight nitpick:

<details>
<summary> Output of <code>./install_sct -h</code> </summary>

```
(venv_sct) joshua@XPS-15-9560:~/repos/spinalcordtoolbox$ ./install_sct -h


*******************************
* Welcome to SCT installation *
*******************************


USAGE
  ./install_sct [-h] [-i] [-y] [-d] [-b] [-v]

OPTIONS
 -h   Show this help
 -i   Install in-place; this is the default when working from git.
 -y   Install without interruption with 'yes' as default answer
 -d   Prevent the (re)-installation of the data/ directory
 -b   Prevent the (re)-installation of the SCT binaries files
 -v   Full verbose



Installation finished successfully!

```
</details>

How much do we care about it printing `Installation finished successfully!`?

Reply:
oh, that's annoying. drat. let's fix that.

Maybe I should move the `trap finish EXIT` line to after the CLI parsing?

Reply:
If this was better structured there would be a `install()` function, and that would run a subshell, and that subshell would define `finish` and run `trap finish EXIT`, not the overall script.

But I think just moving where the trapping happens will fix it.

Reply:
I am super proud of this awk script! awk is underappreciated.

Reply:
FIxed in dd187ba5f3308eda73fe95d40157dd9739e089b8

```
$ ./install_sct -h


*******************************
* Welcome to SCT installation *
*******************************


USAGE
  ./install_sct [-h] [-i] [-y] [-d] [-b] [-v]

OPTIONS
 -h   Show this help
 -i   Install in-place; this is the default when working from git.
 -y   Install without interruption with 'yes' as default answer
 -d   Prevent the (re)-installation of the data/ directory
 -b   Prevent the (re)-installation of the SCT binaries files
 -v   Full verbose



Installation finished successfully!

```

vs

```
$ ./install_sct -h
USAGE
  ./install_sct [-h] [-i] [-y] [-d] [-b] [-v]

OPTIONS
 -h   Show this help
 -i   Install in-place; this is the default when working from git.
 -y   Install without interruption with 'yes' as default answer
 -d   Prevent the (re)-installation of the data/ directory
 -b   Prevent the (re)-installation of the SCT binaries files
 -v   Full verbose

```